src/plot_api/plot_api.js

```diff
@@ -2362,6 +2362,193 @@ exports._guiRestyle = guiEdit(restyle);
 exports._guiRelayout = guiEdit(relayout);
 exports._guiUpdate = guiEdit(update);
 
+// For connecting edited layout attributes to uirevision attrs
+// If no `attr` we use `match[1] + '.uirevision'`
+// Ordered by most common edits first, to minimize our search time
+var layoutUIControlPatterns = [
+    {pattern: /^hiddenlabels/, attr: 'legend.uirevision'},
+    {pattern: /^((x|y)axis\d*)\.((auto)?range|title)/, autofill: true},
+
+    // showspikes and modes include those nested inside scenes
+    {pattern: /axis\d*\.showspikes$/, attr: 'modebar.uirevision'},
+    {pattern: /(hover|drag)mode$/, attr: 'modebar.uirevision'},
+
+    {pattern: /^(scene\d*)\.camera/},
+    {pattern: /^(geo\d*)\.(projection|center)/},
+    {pattern: /^(ternary\d*\.[abc]axis)\.(min|title)$/},
+    {pattern: /^(polar\d*\.(radial|angular)axis)\./},
+    {pattern: /^(mapbox\d*)\.(center|zoom|bearing|pitch)/},
+
+    {pattern: /^legend\.(x|y)$/, attr: 'editrevision'},
+    {pattern: /^(shapes|annotations)/, attr: 'editrevision'},
+    {pattern: /^title$/, attr: 'editrevision'}
+];
```

Comment:
This is a bit annoying, having to explicitly match attribute patterns to `uirevision` attributes, but I thought it was important for example to control all the spikes and hover/drag modes from `modebar.uirevision` and not from all the separate `axis` and `scene.uirevision` attributes. But we might be able to simplify it to only explicitly include patterns when the corresponding `uirevision` attribute is not directly adjacent to the edited attribute or one of its parents...

Reply:
This here is probably not ideal performance-wise, but I kinda like having these patterns declared in block, so :ok_hand: